### PR TITLE
docs: build a module section

### PR DESCRIPTION
**Pull request by dinosk**
Addresses #3883

Signed-off-by: Dinos Kousidis <dinos.kousidis@cern.ch>

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -21,6 +21,8 @@
     architecture/application.rst
     deployment/securing-your-instance.rst
     deployment/monitoring.rst
+    invenio-modules.rst
```
Comment (lnielsen) [inline docs/index.rst]: These has to be reorganised.

File: docs/tutorials/build-a-module.rst [hunk 2]
```diff
@@ -7,3 +7,63 @@
...
+    copyright_by_intergovernmental [True]:
+    superproject [Invenio]:
+    transifex_project [invenio-foo]:
+    extension_class [InvenioFoo]:
+    config_prefix [FOO]:
+
+# How to install it and add it to the application
+
+# module diagram
+
```
Comment (dinosk) [inline docs/tutorials/build-a-module.rst]: Added here some ideas for this page

File: docs/tutorials/build-a-module.rst [hunk 3]
```diff
@@ -7,3 +7,635 @@
...
+.. code-block:: bash
+
+    $ git checkout -b dev
+
+Form, views and templates
+-------------------------
+In this tutorial we'll see how to add data to our Invenio application.
+To accomplish this we will cover several parts of the development process such as:
+
+- How to create a form
```
Comment (ppanero) [inline docs/tutorials/build-a-module.rst]: Dots at the end?

File: docs/tutorials/build-a-module.rst [hunk 4]
```diff
@@ -7,3 +7,635 @@
...
+
+Let's now see our Invenio module in action after it is integrated with our Invenio instance.
+
+First we install our new invenio-foo module:
+
+.. code-block:: console
+
+    $ pipenv install --editable .[all]
+
+Then, if you've followed the steps in the Quickstart guide, you can go to the
```
Comment (ppanero) [inline docs/tutorials/build-a-module.rst]: Link to the Quickstart?

File: docs/tutorials/build-a-module.rst [hunk 5]
```diff
@@ -7,3 +7,635 @@
...
+
+    $ mkvirtualenv my-module-venv
+
+Installing the module is very easy, you just need to go to its root directory
+and `pip install` it:
+
+.. code-block:: bash
+
+    (my-module-venv)$ cd invenio-foo/
+    (my-module-venv)$ pip install --editable .[all]
```
[the remark this answers is not in the document]
Comment (dinosk) [reply docs/tutorials/build-a-module.rst]: pip is still used for modules

File: docs/tutorials/build-a-module.rst [hunk 6]
```diff
@@ -5,5 +5,192 @@
...
+static
+>>>>>>
+If your module contains JavaScript or CSS files, they should go in a folder called ``static``. Also, if you want to group them in bundles,
+you should add a ``bundles.py`` file next to the ``static`` folder.
+
+Module naming conventions
+-------------------------
+
+Invenio modules are standalone independent components that implement some
+functionality used by the rest of the Invenio ecosystem. Modules provide API
```
Comment (ChiaraBi) [inline docs/tutorials/build-a-module.rst]: "some _functionalities"_

File: docs/tutorials/build-a-module.rst [hunk 7]
```diff
@@ -5,5 +5,192 @@
...
+
+A module is usually called:
+
+1. with plural noun, meaning "database (of things)", for example
+   ``invenio-records``, ``invenio-tags``, ``invenio-annotations``,
+
+2. with singular noun, meaning "worker (using things)", for example
+   ``invenio-checker``, ``invenio-editor``.
+
+A module may have split its user interface and REST API interface, for example
```
Comment (ChiaraBi) [inline docs/tutorials/build-a-module.rst]: "The user interface and the REST API interface of a module may be split"

File: docs/tutorials/developing-with-invenio.rst [hunk 8]
```diff
@@ -5,5 +5,463 @@
...
+to have installed the package with its *tests* requirements. If you just want to build the
+documentation, you will only need the *docs* requirements (see the install
+section above). Make sure you are at the root directory
+of the module and run:
+
+.. code-block:: console
+
+    (my-module-venv)$ python setup.py build_sphinx
+
+Open ``docs/_build/html/index.html`` in browser and voilà, the documentation is
```
Comment (ChiaraBi) [inline docs/tutorials/developing-with-invenio.rst]: "in _the_ browser"

File: docs/tutorials/developing-with-invenio.rst [hunk 9]
```diff
@@ -5,5 +5,463 @@
...
+there.
+
+Publishing on GitHub
+--------------------
+Before going further in the tutorial, we can publish your repository to GitHub.
+This allows to integrate a continuous integration system such as TravisCI and allow for
```
Comment (ChiaraBi) [inline docs/tutorials/developing-with-invenio.rst]: "and _allows an_ easy publishing"

File: docs/tutorials/developing-with-invenio.rst [hunk 10]
```diff
@@ -5,5 +5,463 @@
...
+                <li>{{record.title}}</li>
+                {% endfor %}
+            </ol>
+            {% endif %}
+          </div>
+        </div>
+      </div>
+    {% endblock page_body %}
+
+And finally, the `success.html` page in
```
Comment (ChiaraBi) [inline docs/tutorials/developing-with-invenio.rst]: There should be "``" instead of signle `

**Pull request closed without merging**